src/EditorFeatures/Core/Shared/Extensions/SpanExtensions.cs

```diff
@@ -17,8 +18,14 @@ internal static class SpanExtensions
     /// </summary>
     /// <param name="span"></param>
     /// <returns></returns>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
     public static TextSpan ToTextSpan(this Span span)
-        => new(span.Start, span.Length);
+    {
+        // this is a terribly ugly hack.  It depends on the fact that the Span and TextSpan both are just two adjacent
+        // ints, and that neither the editor or roslyn will realistically ever change the layout of these types. If
+        // either does, we will blow up immediately, so this is somewhat ok for us to take a dependency on.
+        return Unsafe.As<Span, TextSpan>(ref span);
```

Comment:
>return Unsafe.As<Span, TextSpan>(ref span); [](http://example.com/codeflow?start=8&length=43)

Yikes! What do you think about adding a unit test to roslyn and a unit test vs-platform that verifies that they didn't change?

Reply:
yes. that would make sense if we want to take this :)